decimal64.go

```diff
@@ -474,3 +498,38 @@ func propagateNan(d ...*decParts) *Decimal64 {
 	}
 	return nil
 }
+
+// Class is a miscellaneous operation that takes one operand and provides the class the decimal is in.
+// This function is formally documented here http://speleotrove.com/decimal/damisc.html#refclass.
+func Class(d Decimal64) string {
+	dp := d.getParts()
+
+	switch dp.fl {
+	case flSNaN:
+		return "sNaN"
+	case flQNaN:
+		return "NaN"
+	}
+
+	res := ""
+	if dp.sign == 1 {
+		res += "-"
+	} else {
+		res += "+"
+	}
+
+	switch dp.fl {
+	case flInf:
```

Comment:
I don't know why you've used another switch case here; You've created methods specifically for decparts and yoi're not using them

Reply:
one of the previous reviews from Marcelo recommended switch statements, but you are right about how code should be independent of other code